physical/oci_objectstorage/oci_objectstorage.go

```diff
@@ -0,0 +1,398 @@
+		if err != nil {
+			return nil, errwrap.Wrapf("failed creating InstancePrincipalConfigurationProvider: {{err}}", err)
+		}
+	}
+
+	objectStorageClient, err := objectstorage.NewObjectStorageClientWithConfigurationProvider(cp)
+	if err != nil {
+		return nil, errwrap.Wrapf("failed creating NewObjectStorageClientWithConfigurationProvider: {{err}}", err)
+	}
+
+	logger.Debug("physical/oci: configuration",
+		"bucketName", bucketName,
+		"namespaceName", namespaceName,
+		"haEnabled", haEnabled,
+		"lockBucketName", lockBucketName,
+		"authTypeAPIKey", authTypeAPIKeyBool,
+	)
+
+	return &Backend{
+		client:         &objectStorageClient,
+		bucketName:     bucketName,
+		logger:         logger,
+		permitPool:     physical.NewPermitPool(MaxNumberOfPermits),
+		namespaceName:  namespaceName,
+		haEnabled:      haEnabled,
+		lockBucketName: lockBucketName,
+	}, nil
+}
+
+func (o *Backend) Put(ctx context.Context, entry *physical.Entry) error {
+	o.logger.Debug("physical/oci: put")
+	defer metrics.MeasureSince(metricPutFull, time.Now())
+	startAcquirePool := time.Now()
+	metrics.SetGauge(metricPermitsUsed, float32(o.permitPool.CurrentPermits()))
+	o.permitPool.Acquire()
+	defer o.permitPool.Release()
+	metrics.MeasureSince(metricPutAcquirePool, startAcquirePool)
+
+	defer metrics.MeasureSince(metricPut, time.Now())
+	size := int64(binary.Size(entry.Value))
+	opcClientRequestId, err := uuid.GenerateUUID()
+	if err != nil {
+		o.logger.Debug("physical/oci: Put: error generating UUID")
+		opcClientRequestId = ""
+	}
+	o.logger.Debug("physical/oci: put", "opc-client-request-id", opcClientRequestId)
+	request := objectstorage.PutObjectRequest{
+		NamespaceName:      &o.namespaceName,
+		BucketName:         &o.bucketName,
+		ObjectName:         &entry.Key,
+		ContentLength:      &size,
+		PutObjectBody:      ioutil.NopCloser(bytes.NewReader(entry.Value)),
+		OpcMeta:            nil,
+		OpcClientRequestId: &opcClientRequestId,
+	}
+
+	resp, err := o.client.PutObject(ctx, request)
+	o.logRequest("put", resp.RawResponse, resp.OpcClientRequestId, resp.OpcRequestId, err)
+
+	if err != nil {
+		metrics.SetGauge(metricPutFailed, 1)
+
+		if resp.RawResponse != nil {
+			defer resp.RawResponse.Body.Close()
+		}
+
+		return errwrap.Wrapf("failed to put data: {{err}}", err)
+	}
+	o.logger.Debug("physical/oci: put end")
+
+	defer resp.RawResponse.Body.Close()
+
+	return nil
+}
+
+func (o *Backend) Get(ctx context.Context, key string) (*physical.Entry, error) {
+	o.logger.Debug("physical/oci: get")
+	defer metrics.MeasureSince(metricGetFull, time.Now())
+	metrics.SetGauge(metricPermitsUsed, float32(o.permitPool.CurrentPermits()))
+	startAcquirePool := time.Now()
+	o.permitPool.Acquire()
+	defer o.permitPool.Release()
+	metrics.MeasureSince(metricGetAcquirePool, startAcquirePool)
+
+	defer metrics.MeasureSince(metricGet, time.Now())
+	opcClientRequestId, err := uuid.GenerateUUID()
+	if err != nil {
+		o.logger.Debug("physical/oci: Get: error generating UUID")
```

Comment:
I'd just return the `err` here. Not being able to get a random number is indicative of low level problems and we'd want to fail the request.